garak/generators/huggingface.py

```diff
@@ -585,6 +584,12 @@ def __init__(self, name="", generations=10, config_root=_config):
             raise ModelNameMissingError(
                 f"Invalid modal name {self.name}, current support: {self.supported_models}."
             )
+
+        from transformers import LlavaNextProcessor, LlavaNextForConditionalGeneration
+
+        PIL = importlib.import_module("PIL")
+        self.Image = PIL.Image
+
```

Comment:
I am a little concerned about importing libs and storing them as attributes in the generator. This will introduce similar issue with `pickle` during multiprocessing that having the client in `OpenAIGenerator` produced.

These might be better served as methods called to load when not already set combined with a `__getstate__()` implementation similar to https://github.com/leondz/garak/blob/3833eefbb29ec10f22298963f668f2b0f99c4526/garak/generators/openai.py#L111-L114

Reply:
I am doing some testing of this pattern in other huggingface changes and while I still like the idea of protecting class for safe pickle support there might be better more work needed in general around how we share generator instances when allowing `mulitprocessing`. As is, shifting heavy lift objects to be created in each new process could be very expensive in terms of resources.

Reply:
I can get this, it makes sense.

On the other hand - if one is doing `multiprocessing` with local models, the consumption of gigabytes/tens of gigabytes of GPU memory per instance seems to shrink the relative difficulties of storing libraries in the generator. 

I am inclined to adopt a safer pattern here, but to not support proactively the general case of parallelisation with locally-run model, instead implementing a `generators.base.Generator` attribute specifying whether a generator is parallelisation-compatible and setting it `False` for 🤗 classes such as `Model` and `Pipeline`. Running parallel local models seems like an edge case - the parallelisation is intended for stuff that's lighter.

Reply:
> implementing a generators.base.Generator attribute specifying whether a generator is parallelisation-compatible and setting it False for 🤗 classes such as Model and Pipeline

This is the approach I am targeting first.

I am also considering looking for methods that can defer multiprocessing approaches for locally executing generators to something specifically provided by the generator, for instance `hugginface` provides the [Accelerate](https://huggingface.co/docs/accelerate/index) library to allow the generator to execute interference using GPUs efficiently and might be reasonable to expose a `_call_model_with_dataset()` or something of the sort that can recieve a set of attempts and have the generator figure out how to parallelize them.

Reply:
@jmartin-tech: heavy externals in these modules now loaded/unloaded using `_load_client` / `_clear_client` pattern